eeps/eep-0064.md

```diff
@@ -0,0 +1,187 @@
+=========
+
+One limitation of Erlang today (June 2023) is the ability to write multi-line
+string binaries. That said, the main reason to consider this EEP is
+[EEP-59][] where documentation attributes can
+benefit from multi-line string binaries: the documentation generates Docs chunks
+format -- documentation in its binary format-- which saves space and makes the
+documentation available to the shell.
+
+Triple-Quoted Binary-String Design Decisions
+---------------------------------------------
+
+### Runtime semantics
+
+Triple-quoted strings should only produce binary-strings. This makes easy to
+avoid typing issues with union types, and makes clear that a triple-quoted string
+does not need function with guards to test if the produce result is a list or a
+binary (this can happen if Erlang adds string interpolation,
+[EEP-62][],
+[PR-7343][]).
+
+Triple-quoted binary-strings do not allow string interpolation and, if we were
+to accept them, only calls to statically known values should be allowed, e.g.,
+macros. This design decision forbids the creation of dynamic documentation that
+generates code at runtime, e.g., doing an IO call to get a value not known
+statically and based on it perform an action (notice that this is allowed when
+using string interpolation).
+
+### Binary-String Design
+
+To make the usage of binary-string similar to other languages, this EEP proposal
+makes the binary-string design decisions similar to the ones for the Elixir language.
+
+#### Binary-String Start and End
+
+Binary-strings must start and end with triple-quotes in their own lines:
+
+*Code Example*
+
+    X = """
+        This is the beginning of the triple-quote text
+        """
+
+*Documentation Example*
+
+    -doc """
+         Removes double-quotes (") from a given string
+         """
+    remove_double_quotes(X) -> 
+
+#### Binary-Strings Closing of Triple-Quotes
+
+The closing of the triple-quotes denotes the space (indentation):
+
+*1. Code Example*
+
+    >> X = """
+           This text
+           has no indentation
+           """
+    
+    Equivalent to: <<"This text\nhas no indentation\n">>
+
+*2. Code Example*
+
+    >> Y = """
+           This text
+           has indentation
+    """
+    
+    Equivalent to: <<"       This text\n       has indentation\n">>
+
+*3. Code Example*
+
+    >> Z = """
+      This text has
+      two space indentation
+    """
+    
+    Equivalent to: <<"  This text has\n  two space indentation\n">>
+
+*1. Documentation Example - No indentation*
+
+    -doc """
+         Removes double-quotes (") from a given string
+         """
+    remove_double_quotes(X) -> 
+
+*2. Documentation Example - No indentation*
+
+    -doc """
+    Removes double-quotes (") from a given string
+    """
+    remove_double_quotes(X) -> 
+
+*3. Documentation Example - Indentation*
+
+    -doc """
+         Removes double-quotes (") from a given string
+    """
+    remove_double_quotes(X) -> 
+
+#### Binary-Strings Errors
```

Comment:
Perhaps worth mentioning it also errors when there is a `"""` and it is not immediately followed spaces and a newline? Elixir says:

```
iex(1)> """foo"""
** (SyntaxError) iex:1:1: heredoc allows only optional whitespace followed by a new line after """
```

Reply:
Since we are still compatible with existing code, and one can write strings using single quotes, we may have to allow code that for some reason is `"""foo"""`, which in Erlang produces `"foo"`. If we use backtick instead of quotes, then your point stands

Reply:
Perhaps it is worth pushing a deprecation warning to Erlang/OTP 26.x that warns on triple quotes?